pkg/issuer/acme/http/http.go

```diff
@@ -103,8 +123,22 @@ func (s *Solver) Present(ctx context.Context, issuer v1.GenericIssuer, ch *cmacm
 	if svcErr != nil {
 		return utilerrors.NewAggregate([]error{podErr, svcErr})
 	}
-	_, ingressErr := s.ensureIngress(ctx, ch, svc.Name)
-	return utilerrors.NewAggregate([]error{podErr, svcErr, ingressErr})
+
+	var ingressErr error
+	if ch.Spec.Solver.HTTP01.Ingress != nil {
+		_, ingressErr = s.ensureIngress(ctx, ch, svc.Name)
+	}
+
+	var istioErr error
+	if ch.Spec.Solver.HTTP01.Istio != nil {
+		if s.IstioEnabled {
+			_, istioErr = s.ensureIstio(ctx, ch, svc.Name)
+		} else {
+			istioErr = errors.New("Istio support was not detected on startup, try restarting cert-manager")
```

Comment:
I suppose it must be possible to watch for the availability  or absence of certain APIs and then start / stop the VirtualService lister as necessary.
Maybe add a TODO comment with a link to an Issue. It'd be nice to not have to worry about the order of installing Istio and cert-manager.

Reply:
I added a TODO comment.

Reply:
Thanks.